diff --git a/src/__pycache__/trainer.cpython-310.pyc b/src/__pycache__/trainer.cpython-310.pyc
index f0373ef..3d90c3a 100644
Binary files a/src/__pycache__/trainer.cpython-310.pyc and b/src/__pycache__/trainer.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/base.cpython-310.pyc b/src/optimizers/__pycache__/base.cpython-310.pyc
index 3bff240..c843bb3 100644
Binary files a/src/optimizers/__pycache__/base.cpython-310.pyc and b/src/optimizers/__pycache__/base.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc b/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc
index 30ceadd..3c9f5a4 100644
Binary files a/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc and b/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc b/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc
index 7e7acce..ba9f562 100644
Binary files a/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc and b/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc b/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc
index 88b87a5..c601206 100644
Binary files a/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc and b/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc differ
diff --git a/src/optimizers/base.py b/src/optimizers/base.py
index ab12c7a..db101ff 100644
--- a/src/optimizers/base.py
+++ b/src/optimizers/base.py
@@ -5,6 +5,7 @@ import torch
 import numpy as np
 from .opt_utils import VectorSampler
 from gradient_pruning import fast_random_mask_like
+from torch.optim import SGD
 
 class ZeroOrderOptimizer(Optimizer, ABC):
     def __init__(self,
@@ -64,6 +65,21 @@ class ZeroOrderOptimizer(Optimizer, ABC):
         self.zo_eps = self._calculate_zo_eps(eps=eps)
         # print("DONE init")
 
+        self._inner_optimizers = []
+        for group in self.param_groups:
+            self._inner_optimizers.append(
+                SGD(group['params'], lr=group['lr'], momentum=group['momentum'])
+            )
+       
+        self._lr_schedulers = None
+
+    def set_lr_schedulers(self, lr_schedulers: List):
+        """        
+        Args:
+            lr_schedulers: list of schedulers
+        """
+        self._lr_schedulers = lr_schedulers
+
     def _prepare_parameters(self) -> None:
         """Prepares parameters for optimization. Common for all optimizer's steps"""
         self.named_parameters_to_optim = [
diff --git a/src/optimizers/jaguar_muon.py b/src/optimizers/jaguar_muon.py
index e59b3c9..fb4d30e 100644
--- a/src/optimizers/jaguar_muon.py
+++ b/src/optimizers/jaguar_muon.py
@@ -1,6 +1,6 @@
 from .base import ZeroOrderOptimizer
 import torch
-from torch.optim import SGD
+# from torch.optim import SGD
 import numpy as np
 from typing import Optional, Callable, Dict, Any, Union, List, Iterable, Tuple
 
@@ -29,27 +29,13 @@ class Jaguar_MUON(ZeroOrderOptimizer):
         self.beta = beta
         self.use_smoothing = use_smoothing
 
-        self._inner_optimizers = []
-        for group in self.param_groups:
-            self._inner_optimizers.append(
-                SGD(group['params'], lr=group['lr'], momentum=group['momentum'])
-            )
+        # self._inner_optimizers = []
+        # for group in self.param_groups:
+        #     self._inner_optimizers.append(
+        #         SGD(group['params'], lr=group['lr'], momentum=group['momentum'])
+        #     )
        
-        self._lr_schedulers = None
-
-    def set_lr_schedulers(self, lr_schedulers: List):
-        """
-        Устанавливает scheduler'ы для внутренних оптимизаторов.
-        
-        Args:
-            lr_schedulers: Список scheduler'ов, по одному на каждую группу параметров
-        """
-        if len(lr_schedulers) != len(self.param_groups):
-            raise ValueError(
-                f"Количество scheduler'ов ({len(lr_schedulers)}) должно "
-                f"соответствовать количеству групп параметров ({len(self.param_groups)})"
-            )
-        self._lr_schedulers = lr_schedulers
+        # self._lr_schedulers = None
             
     @torch.no_grad()
     def step(self, closure=None):
@@ -158,7 +144,7 @@ class Jaguar_MUON(ZeroOrderOptimizer):
         for name, param in self.named_parameters_to_optim:
             param.grad = original_grads[name]
 
-        for group_idx, _ in enumerate(self.param_groups):
-            self._lr_schedulers[group_idx].step()
+        # for group_idx, _ in enumerate(self.param_groups):
+        #     self._lr_schedulers[group_idx].step()
 
         return loss1
diff --git a/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc b/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc
index 5ed94fb..d9c92b2 100644
Binary files a/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc and b/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc differ
diff --git a/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc b/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc
index 6f12749..1549d17 100644
Binary files a/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc and b/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc differ
diff --git a/src/run_script.sh b/src/run_script.sh
index de74c3e..98a9852 100755
--- a/src/run_script.sh
+++ b/src/run_script.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-export CUDA_VISIBLE_DEVICES=2,3
+export CUDA_VISIBLE_DEVICES=0,3
 export WANDB_DISABLED="false"          
 export WANDB_ENTITY="andrey"   
 export WANDB_PROJECT="zo-lib-run"    
diff --git a/src/trainer.py b/src/trainer.py
index 238cd2e..f84a04c 100644
--- a/src/trainer.py
+++ b/src/trainer.py
@@ -358,12 +358,6 @@ class OurTrainer(Trainer):
             self.optimizer = ZO_SamplingMUON(self.model.parameters(), tau=1e-1, lr=1e-3, eps=1e-2, momentum=0.0, gradient_sparsity=self.gradient_sparsity)
         elif args.trainer == "jaguar_muon":
             self.optimizer = Jaguar_MUON(self.model.parameters(), tau=1e-1, beta=1e-2, use_smoothing=True, lr=1e-3, eps=1e-2, momentum=0.0, gradient_sparsity=self.gradient_sparsity)
-            inner_optimizers = optimizer._inner_optimizers
-            schedulers = [
-                StepLR(opt, step_size=30, gamma=0.1) 
-                for opt in inner_optimizers
-            ]
-            optimizer.set_lr_schedulers(schedulers)
         else:
             # assert args.lr_scheduler_type == 'constant', "we did not implement lr_schedule."
             if args.optimizer == "adam": # FIXME: what to do with this? 
@@ -372,6 +366,12 @@ class OurTrainer(Trainer):
                 self.optimizer = SGD(self.model.parameters(), lr=args.learning_rate, momentum=args.momentum)
             else: 
                 raise NotImplementedError(f"Optimizer {args.optimizer} is not implemented")
+        inner_optimizers = self.optimizer._inner_optimizers
+        schedulers = [
+            get_scheduler(opt, scheduler_type=args.scheduler, num_training_steps=args.num_training_steps, warmup_steps=args.warmup_steps, min_lr_ratio=args.min_lr_ratio) 
+            for opt in inner_optimizers
+        ]
+        self.optimizer.set_lr_schedulers(schedulers)
 
         # self.scheduler = CosineAnnealingLR(self.optimizer, T_max=self.args.max_steps, eta_min=1e-8)
         # important: at this point:
@@ -539,7 +539,7 @@ class OurTrainer(Trainer):
                 if args.trainer == "jaguar_muon":            
                     for scheduler in schedulers:
                         scheduler.step()
-                self.scheduler.step()
+                # self.scheduler.step()
                 print(f"Step {total_steps}, LR: {self.optimizer.param_groups[0]['lr']:.2e}")
 
                 if (
diff --git a/src/wandb/debug-internal.log b/src/wandb/debug-internal.log
index 7c3cfcd..66c3ddd 120000
--- a/src/wandb/debug-internal.log
+++ b/src/wandb/debug-internal.log
@@ -1 +1 @@
-run-20250717_142655-f14rxscu/logs/debug-internal.log
\ No newline at end of file
+run-20250805_141006-jnzb1jsx/logs/debug-internal.log
\ No newline at end of file
diff --git a/src/wandb/debug.log b/src/wandb/debug.log
index 1f4632d..f1fff63 120000
--- a/src/wandb/debug.log
+++ b/src/wandb/debug.log
@@ -1 +1 @@
-run-20250717_142655-f14rxscu/logs/debug.log
\ No newline at end of file
+run-20250805_141006-jnzb1jsx/logs/debug.log
\ No newline at end of file
diff --git a/src/wandb/latest-run b/src/wandb/latest-run
index caaf7bb..2c3d1e4 120000
--- a/src/wandb/latest-run
+++ b/src/wandb/latest-run
@@ -1 +1 @@
-run-20250717_142655-f14rxscu
\ No newline at end of file
+run-20250805_141006-jnzb1jsx
\ No newline at end of file
