diff --git a/src/__pycache__/trainer.cpython-310.pyc b/src/__pycache__/trainer.cpython-310.pyc
index f0373ef..5f1f71b 100644
Binary files a/src/__pycache__/trainer.cpython-310.pyc and b/src/__pycache__/trainer.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/base.cpython-310.pyc b/src/optimizers/__pycache__/base.cpython-310.pyc
index 3bff240..3243fe5 100644
Binary files a/src/optimizers/__pycache__/base.cpython-310.pyc and b/src/optimizers/__pycache__/base.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc b/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc
index 30ceadd..148489b 100644
Binary files a/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc and b/src/optimizers/__pycache__/jaguar_muon.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc b/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc
index 7e7acce..ba9f562 100644
Binary files a/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc and b/src/optimizers/__pycache__/jaguar_signsgd.cpython-310.pyc differ
diff --git a/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc b/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc
index 88b87a5..c601206 100644
Binary files a/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc and b/src/optimizers/__pycache__/zo_sgd.cpython-310.pyc differ
diff --git a/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc b/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc
index 5ed94fb..d9c92b2 100644
Binary files a/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc and b/src/optimizers/opt_utils/__pycache__/__init__.cpython-310.pyc differ
diff --git a/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc b/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc
index 6f12749..1549d17 100644
Binary files a/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc and b/src/optimizers/opt_utils/__pycache__/vector_sampling.cpython-310.pyc differ
diff --git a/src/run_script.sh b/src/run_script.sh
index de74c3e..027b529 100755
--- a/src/run_script.sh
+++ b/src/run_script.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-export CUDA_VISIBLE_DEVICES=2,3
+export CUDA_VISIBLE_DEVICES=3
 export WANDB_DISABLED="false"          
 export WANDB_ENTITY="andrey"   
 export WANDB_PROJECT="zo-lib-run"    
diff --git a/src/trainer.py b/src/trainer.py
index 238cd2e..e82fda9 100644
--- a/src/trainer.py
+++ b/src/trainer.py
@@ -358,12 +358,12 @@ class OurTrainer(Trainer):
             self.optimizer = ZO_SamplingMUON(self.model.parameters(), tau=1e-1, lr=1e-3, eps=1e-2, momentum=0.0, gradient_sparsity=self.gradient_sparsity)
         elif args.trainer == "jaguar_muon":
             self.optimizer = Jaguar_MUON(self.model.parameters(), tau=1e-1, beta=1e-2, use_smoothing=True, lr=1e-3, eps=1e-2, momentum=0.0, gradient_sparsity=self.gradient_sparsity)
-            inner_optimizers = optimizer._inner_optimizers
+            inner_optimizers = self.optimizer._inner_optimizers
             schedulers = [
-                StepLR(opt, step_size=30, gamma=0.1) 
+                get_scheduler(opt, scheduler_type=args.scheduler, num_training_steps=args.num_training_steps, warmup_steps=args.warmup_steps, min_lr_ratio=args.min_lr_ratio) 
                 for opt in inner_optimizers
             ]
-            optimizer.set_lr_schedulers(schedulers)
+            self.optimizer.set_lr_schedulers(schedulers)
         else:
             # assert args.lr_scheduler_type == 'constant', "we did not implement lr_schedule."
             if args.optimizer == "adam": # FIXME: what to do with this? 
diff --git a/src/wandb/debug-internal.log b/src/wandb/debug-internal.log
index 7c3cfcd..d074ec5 120000
--- a/src/wandb/debug-internal.log
+++ b/src/wandb/debug-internal.log
@@ -1 +1 @@
-run-20250717_142655-f14rxscu/logs/debug-internal.log
\ No newline at end of file
+run-20250804_144537-0q88tm9l/logs/debug-internal.log
\ No newline at end of file
diff --git a/src/wandb/debug.log b/src/wandb/debug.log
index 1f4632d..55d9d38 120000
--- a/src/wandb/debug.log
+++ b/src/wandb/debug.log
@@ -1 +1 @@
-run-20250717_142655-f14rxscu/logs/debug.log
\ No newline at end of file
+run-20250804_144537-0q88tm9l/logs/debug.log
\ No newline at end of file
diff --git a/src/wandb/latest-run b/src/wandb/latest-run
index caaf7bb..310402d 120000
--- a/src/wandb/latest-run
+++ b/src/wandb/latest-run
@@ -1 +1 @@
-run-20250717_142655-f14rxscu
\ No newline at end of file
+run-20250804_144537-0q88tm9l
\ No newline at end of file
